components/utils/highCharts.ts

```diff
@@ -10,13 +13,33 @@ export const plotLinesColor = "#FF0000"; // red;
 export const timeSeriesColors = originalHighchartsColors!.slice(1);
 
 // Colours from Bang Wong palette, see https://davidmathlogic.com/colorblind
-const colorBlindSafeColors = [
+export const colorBlindSafeColors = [
   { name: "Vermillion", hex: "#d55e00" },
   { name: "Bluish green", hex: "#009e73" },
   { name: "Sky blue", hex: "#56b4e9" },
 ];
 
-export const costsPieColors = ["rgba(1,1,1,0)"].concat(Array.from<string>(colorBlindSafeColors.map(color => color.hex))); // Make the center circle translucent
+// Create a range of N colors varying in brightness
+// NB This won't work well with colors that are already very light or very dark.
+export const getColorVariants = (colorHex: string, numberOfVariants: number) => {
+  const rgb = hexRgb(colorHex);
+  const colors = [];
+  const minBrightness = 0.75; // 75% brightness
+  const maxBrightness = 1.25; // 125% brightness
+  for (let i = 0; i < numberOfVariants; i++) {
+    const factor = i / (numberOfVariants - 1);
+    const brightness = minBrightness + (maxBrightness - minBrightness) * factor;
+    let r = Math.round(rgb.red * brightness);
+    let g = Math.round(rgb.green * brightness);
+    let b = Math.round(rgb.blue * brightness);
+    r = Math.max(0, Math.min(255, r));
+    g = Math.max(0, Math.min(255, g));
+    b = Math.max(0, Math.min(255, b));
```

Comment:
This is potentially going to change the hue of the variants isn't it? e.g. if you start off with an r, g or b value which gets clipped at its max while the others don't?
Would it make more sense to work in hsl? Maybe using [this](https://www.npmjs.com/package/color-convert) if we haven't already got a converter which will do it.

Reply:
What a good idea

Reply:
I used your hsl idea, and here it is working:
![image](https://github.com/user-attachments/assets/7efb058b-2ac3-46a7-b3c2-5e80990b2a52)

Reply:
Ha, these variants seem a bit blander, I think I preferred them before! :facepalm: Up to you!